.github/workflows/cd.yml

```diff
@@ -0,0 +1,32 @@
+name: cd
+on:
+  workflow_run:
+    workflows: ["ci"]
+    branches-ignore: [main]
+    types:
+    - completed
+  push:
+    tags:
+    - "v[0-9]+.[0-9]+.[0-9]+"
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@dc323e67f16fb5f7663d20ff7941f27f5809e9b6 # v3.2.0
+      with:
+        fetch-depth: 0
+
+    - name: cd/build-docker
+      run: make build-image
+
+    - name: cd/docker-push
+      run: TAG=${GITHUB_REF_NAME} ./scripts/push-docker.sh
```

Comment:
This will build and override the image with the branch name everytime you push a new commit on the open PR . 
Is this the expected behaviour ? 

You should consider removing the `workflow_run`directive

Reply:
This is ok for me, so we don't bloat everything with an image per commit, but image caching could be an issue while testing I assume. cc: @spirosoik @gabrieljackson

Reply:
This was always like this actually. It's up to you guys, we can change to SHA if you need that granularity.

Reply:
I think we can move forward with the branch name for now.